pkg/handler/common/cluster.go

```diff
@@ -1028,15 +1028,16 @@ func convertInternalCCMStatusToExternal(cluster *kubermaticv1.Cluster, datacente
 
 	if cluster.Spec.Features[kubermaticv1.ClusterFeatureExternalCloudProvider] {
 		status.ExternalCCM = true
-		status.MigrationNeeded = nil
 	} else if cloudcontroller.ExternalCloudControllerFeatureSupported(datacenter, cluster) {
 		status.MigrationNeeded = pointer.BoolPtr(true)
 	}
 
 	if _, condition := helper.GetClusterCondition(cluster, kubermaticv1.ClusterConditionCSIKubeletMigrationCompleted); condition != nil {
-		status.MigrationCompleted = pointer.BoolPtr(helper.ClusterConditionHasStatus(cluster, kubermaticv1.ClusterConditionCSIKubeletMigrationCompleted, corev1.ConditionTrue))
-	} else {
-		status.MigrationCompleted = nil
+		if helper.ClusterConditionHasStatus(cluster, kubermaticv1.ClusterConditionCSIKubeletMigrationCompleted, corev1.ConditionTrue) {
+			status.MigrationCompleted = pointer.BoolPtr(true)
+		} else {
+			status.MigrationInProgress = pointer.BoolPtr(true)
+		}
```

Comment:
This might produce an incorrect result between starting the migration and the controller. For example:

* The ExternalCCM is not enabled, but it's supported. We'll set `MigrationNeeded` to `true`. Other fields will be `nil`.
* We enable the CCM/CSI migration. At this point, the feature is enabled, so we'll set `ExternalCCM` to `true`. We will not set `MigrationNeeded`.
* The cluster condition is not set because the controller is still not enabled. This will cause other fields to be `nil`, so we will have only `ExternalCCM` set to `true`. This is not expected and will report a wrong status in the UI.

I think that this can be solved by checking the annotations:

* `ExternalCCM` will be `true` if the cluster feature is enabled
* `MigrationNeeded` will be `true` if the cluster feature is disabled, but the provider supports the external CCM
  * It'll become `false` as soon as we start the migration
  * If it's `true`, we don't need to set the other field
 * `MigrationInProgress` will be `true` if the cluster feature is enabled, and there are the CCM/CSI migration annotations, and the cluster condition is `nil` or `false`. Otherwise, it'll be `false`.
 * `MigrationComplete` will be `true` if the cluster condition is true
   * I'm not sure if this field is really needed and useful, since I don't think that we need to show this information in the UI. Anyways, let's keep it for now and see later.